lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1667,6 +1667,60 @@ struct HandshakeAnalysisPass
 };
 } // namespace
 
+// Temporary: remove unused function args
+// The array arguements are impelemented as local memory ops in handshake
+// dialect, and the original arguements have no use. This pass temporary removes
+// these used args to let the simulator work properly by reading external data
+// files.
+namespace {
+
+struct HandshakeCleanFuncArgPass
+    : public HandshakeCleanFuncArgBase<HandshakeCleanFuncArgPass> {
+  void runOnOperation() override {
+    auto m = getOperation();
+    for (auto funcOp : m.getOps<handshake::FuncOp>()) {
+      OpBuilder rewriter(funcOp);
+
+      // Get function arguments
+      std::vector<unsigned int> argIndices;
+      llvm::SmallVector<mlir::Type, 8> argTypes;
+      auto i = 0;
+      for (auto &arg : funcOp.getArguments()) {
+        auto type = arg.getType();
+        if (arg.use_empty())
+          argIndices.push_back(i);
+        else if (arg.hasOneUse()) {
```

Comment:
Style-wise, instead of nesting if-else like this, you might be able to simplify the logic. Something like:

```
if (arg.use_empty()) {
  argIndices.push_back(i)
  continue;
}

if (auto sinkOp = ...) {
  sinkOp.erase();
  argIndices.push_back(i);
  continue;
}

argTypes.push_back(type);
++i;
```